### ISSUE TITLE
Are USB HDD's supported?

### ISSUE THREAD

**Issue by Abscissa** [papylhomme/diskmonitor#19]
On the one I tried, diskmonitor just said the drive's status was unknown. Wasn't sure if this was a diskmonitor limitation or some kind of driver problem, etc, on my end.

**Comment by papylhomme** [comment]
This project relies entirely on UDisks (which itself relies on libsmart), the unknown status indicates that UDisks doesn't provide SMART information for this drive.

You can check this using a DBus client (`qdbusviewer` on KDE), search for UDisk2 in the **system bus**, and lookup your drive under `org/freedesktop/UDisk2/drives`. If it provides a `org.freedesktop.UDisk2.Drive.Ata` interface but doesn't show in diskmonitor then it's a bug.

To my knowledge, most USB HDD only provide the `org.freedesktop.UDisk2.Drive` interface, as they might need a specific driver to query SMART information. See https://www.smartmontools.org/wiki/USB and https://www.smartmontools.org/wiki/Supported_USB-Devices

You can try using the command `smartctl -a /dev/xxx` to check if the drive provides them directly or not. If you need to specify a driver using the option `-d` then UDisks2 won't be able to infer this itself.

As a side note, there might be some DB or config files to handwrite this information for libsmart, if you're interested enough it's worse looking for it, let me know if you find something.

**Comment by papylhomme** [comment]
I investigated this a bit further, and UDisks2 use `libatasmart`, which is not related to `smartmontools`, so sorry for my previous misleading answer.

As stated in the project's [README](https://github.com/Rupan/libatasmart)

> USB disks generally don't support SAT and thus generally cannot be queried for S.M.A.R.T. data.

so it will depend on your particular disk. I tested with an external SSD, and everything worked as expected in diskmonitor (interestingly it wasn't recognized by smartmontools...).
